source/view/index.js

```diff
@@ -529,6 +538,15 @@ function runBlockly() {
 
 function setActiveBlocklyTab() {
     if ( currentBlocklyNodeID !== this.id ) {
+        
+        var blocklyNode = blocklyNodes[ this.id ];
+
+        if ( Blockly.mainWorkspace ) {
+            Blockly.mainWorkspace.maxBlocks = blocklyNode.ramMax;
+        } else {
+            console.log("Blockly.mainWorkspace not found!");
+        }
+
         vwf_view.kernel.setProperty( appID, "blockly_activeNodeID", this.id );
```

Comment:
Since `blockly_activeNodeID` is being set here and `maxBlocks` is being set in `satProperty`, we should be able leave `setActiveBlocklyTab` as it was.